### ISSUE TITLE
How can one evaluate whether the parameter settings for eqs and scales are appropriate?

### ISSUE THREAD

**Issue by GEOSCIENCELXG** [tobiscode/disstans#8]
I tried different sets of parameters: eqs and scale, get different results with the code lines:

    # make a single solution
    rw_func = disstans.solvers.InverseReweighting(eps=1e-7, scale=1e-6)
    print(rw_func)

    stats = net.spatialfit("final",
                           penalty=[100, 100, 50],
                           spatial_l0_models=["Transient"],
                           spatial_reweight_iters=15,
                           reweight_func=rw_func,
                           formal_covariance=True,
                           use_data_covariance=True,
                           verbose=True,
                           extended_stats=True,
                           keep_mdl_res_as=("model_srw", "resid_srw"))

for:   **eps=1e-7, scale=1e-6**
![example_1e_scalo_scale1e-6eps1e-7](https://github.com/user-attachments/assets/2a458b40-f2f7-4896-b696-d3e32079b665)

for:   **eps=1e-6, scale=1e-5**
![example_1e_scalo_scale1e-5eps1e-6](https://github.com/user-attachments/assets/e838a709-e1f2-4647-a4fd-9198926ea57a)

for:   **eps=1e-6, scale=1e-4**
![example_1e_scalo_scale1e-4eps1e-6](https://github.com/user-attachments/assets/3291cc56-7622-47ce-876d-0edf0c94343d)

for:   **eps=1e-6, scale=1e-3**
![example_1e_scalo_scale1e-3eps1e-6](https://github.com/user-attachments/assets/ce648b2a-64a8-4884-b572-bf5eff12dbb4)

the bellow print info for:   **eps=1e-7, scale=1e-6**
<disstans.solvers.InverseReweighting object at 0x7fa77e510770>
Calculating scale lengths
Distance percentiles in km (5-50-95): [11.1, 54.5, 153.7]
Initial fit
Fitting station models: 100%|#############################################| 23/23 [00:03<00:00,  7.30station/s]
Number of reweighted non-zero parameters: 69/17388
Number of unique reweighted non-zero parameters per component: [15, 8, 4]
Evaluating station models: 100%|##########################################| 23/23 [00:03<00:00,  6.43station/s]
Updating weights
Stacking model Transient
Weight percentiles (5-50-95): [2.865637454, 9.9999482187, 9.9999998983]
Fit after 1 reweightings
Fitting station models: 100%|#############################################| 23/23 [00:03<00:00,  6.50station/s]
Number of reweighted non-zero parameters: 540/17388
Number of unique reweighted non-zero parameters per component: [105, 79, 22]
RMS difference of 'Transient' parameters = 4.6578046332 (535 changed)
Evaluating station models: 100%|##########################################| 23/23 [00:03<00:00,  6.43station/s]
Updating weights
Stacking model Transient
Weight percentiles (5-50-95): [0.12258819708, 3.3702329783, 9.9998059108]
Fit after 2 reweightings
Fitting station models: 100%|#############################################| 23/23 [00:03<00:00,  6.66station/s]
Number of reweighted non-zero parameters: 1334/17388
Number of unique reweighted non-zero parameters per component: [128, 107, 48]
RMS difference of 'Transient' parameters = 19.502297755 (1234 changed)
Evaluating station models: 100%|##########################################| 23/23 [00:03<00:00,  6.36station/s]
Updating weights
Stacking model Transient
Weight percentiles (5-50-95): [2.4127979954e-06, 7.4514125217, 9.9887794806]
Fit after 3 reweightings
Fitting station models: 100%|#############################################| 23/23 [00:03<00:00,  6.89station/s]
Number of reweighted non-zero parameters: 1435/17388
Number of unique reweighted non-zero parameters per component: [55, 50, 37]
RMS difference of 'Transient' parameters = 727.30320756 (1193 changed)
Evaluating station models: 100%|##########################################| 23/23 [00:03<00:00,  6.19station/s]
Updating weights
Stacking model Transient
Weight percentiles (5-50-95): [8.599185417e-07, 9.9796836164, 9.9999719666]
Fit after 4 reweightings
Fitting station models: 100%|#############################################| 23/23 [00:03<00:00,  6.63station/s]
Number of reweighted non-zero parameters: 1487/17388
Number of unique reweighted non-zero parameters per component: [36, 33, 30]
RMS difference of 'Transient' parameters = 892.23663502 (416 changed)
Evaluating station models: 100%|##########################################| 23/23 [00:03<00:00,  6.61station/s]
Updating weights
Stacking model Transient
Weight percentiles (5-50-95): [6.1410992872e-07, 9.9994786381, 9.9999999883]
Fit after 5 reweightings
Fitting station models: 100%|#############################################| 23/23 [00:03<00:00,  6.81station/s]
Number of reweighted non-zero parameters: 1495/17388
Number of unique reweighted non-zero parameters per component: [34, 26, 25]
RMS difference of 'Transient' parameters = 660.74164374 (126 changed)
Evaluating station models: 100%|##########################################| 23/23 [00:03<00:00,  7.34station/s]
Updating weights
Stacking model Transient
Weight percentiles (5-50-95): [6.3724592609e-07, 9.9998055219, 9.9999999238]
Fit after 6 reweightings
Fitting station models: 100%|#############################################| 23/23 [00:03<00:00,  6.65station/s]
Number of reweighted non-zero parameters: 1501/17388
Number of unique reweighted non-zero parameters per component: [34, 26, 25]
RMS difference of 'Transient' parameters = 37.236371436 (12 changed)
Evaluating station models: 100%|##########################################| 23/23 [00:03<00:00,  6.08station/s]
Updating weights
Stacking model Transient
Weight percentiles (5-50-95): [6.247084561e-07, 9.9800669906, 9.9999997301]
Fit after 7 reweightings
Fitting station models: 100%|#############################################| 23/23 [00:03<00:00,  6.74station/s]
Number of reweighted non-zero parameters: 1505/17388
Number of unique reweighted non-zero parameters per component: [34, 26, 25]
RMS difference of 'Transient' parameters = 28.57203359 (4 changed)
Evaluating station models: 100%|##########################################| 23/23 [00:03<00:00,  6.85station/s]
Updating weights
Stacking model Transient
Weight percentiles (5-50-95): [6.1582308377e-07, 9.9832787962, 9.9999999584]
Fit after 8 reweightings
Fitting station models: 100%|#############################################| 23/23 [00:03<00:00,  6.78station/s]
Number of reweighted non-zero parameters: 1505/17388
Number of unique reweighted non-zero parameters per component: [34, 26, 25]
RMS difference of 'Transient' parameters = 0.46177523358 (0 changed)
Evaluating station models: 100%|##########################################| 23/23 [00:03<00:00,  7.38station/s]
Updating weights
Stacking model Transient
Weight percentiles (5-50-95): [6.1582311316e-07, 9.9802809461, 9.9999999085]
Fit after 9 reweightings
Fitting station models: 100%|#############################################| 23/23 [00:03<00:00,  7.03station/s]
Number of reweighted non-zero parameters: 1506/17388
Number of unique reweighted non-zero parameters per component: [34, 26, 25]
RMS difference of 'Transient' parameters = 0.28509152875 (1 changed)
Evaluating station models: 100%|##########################################| 23/23 [00:03<00:00,  7.33station/s]
Updating weights
Stacking model Transient
Weight percentiles (5-50-95): [6.1582317735e-07, 9.9787527599, 9.9999999641]
Fit after 10 reweightings
Fitting station models: 100%|#############################################| 23/23 [00:03<00:00,  7.36station/s]
Number of reweighted non-zero parameters: 1510/17388
Number of unique reweighted non-zero parameters per component: [35, 27, 25]
RMS difference of 'Transient' parameters = 1.4496710442 (4 changed)
Evaluating station models: 100%|##########################################| 23/23 [00:03<00:00,  6.46station/s]
Updating weights
Stacking model Transient
Weight percentiles (5-50-95): [6.1798379472e-07, 9.9857494305, 9.999999821]
Fit after 11 reweightings
Fitting station models: 100%|#############################################| 23/23 [00:03<00:00,  6.58station/s]
Number of reweighted non-zero parameters: 1508/17388
Number of unique reweighted non-zero parameters per component: [34, 27, 25]
RMS difference of 'Transient' parameters = 0.58703316112 (4 changed)
Evaluating station models: 100%|##########################################| 23/23 [00:03<00:00,  6.54station/s]
Updating weights
Stacking model Transient
Weight percentiles (5-50-95): [6.1798379579e-07, 9.9856397767, 9.9999999086]
Fit after 12 reweightings
Fitting station models: 100%|#############################################| 23/23 [00:03<00:00,  6.89station/s]
Number of reweighted non-zero parameters: 1507/17388
Number of unique reweighted non-zero parameters per component: [34, 27, 25]
RMS difference of 'Transient' parameters = 0.21537826153 (1 changed)
Evaluating station models: 100%|##########################################| 23/23 [00:03<00:00,  6.75station/s]
Updating weights
Stacking model Transient
Weight percentiles (5-50-95): [6.1798380068e-07, 9.9815902386, 9.999999902]
Fit after 13 reweightings
Fitting station models: 100%|#############################################| 23/23 [00:03<00:00,  7.20station/s]
Number of reweighted non-zero parameters: 1507/17388
Number of unique reweighted non-zero parameters per component: [35, 27, 25]
RMS difference of 'Transient' parameters = 0.56750215874 (2 changed)
Evaluating station models: 100%|##########################################| 23/23 [00:03<00:00,  7.28station/s]
Updating weights
Stacking model Transient
Weight percentiles (5-50-95): [6.1926201271e-07, 9.9818730216, 9.9999996052]
Fit after 14 reweightings
Fitting station models: 100%|#############################################| 23/23 [00:03<00:00,  6.31station/s]
Number of reweighted non-zero parameters: 1511/17388
Number of unique reweighted non-zero parameters per component: [35, 27, 25]
RMS difference of 'Transient' parameters = 0.79585755606 (4 changed)
Evaluating station models: 100%|##########################################| 23/23 [00:03<00:00,  6.52station/s]
Updating weights
Stacking model Transient
Weight percentiles (5-50-95): [6.1798379475e-07, 9.9862034927, 9.9999999188]
Fit after 15 reweightings
Fitting station models: 100%|#############################################| 23/23 [00:03<00:00,  6.94station/s]
Number of reweighted non-zero parameters: 1506/17388
Number of unique reweighted non-zero parameters per component: [34, 26, 25]
RMS difference of 'Transient' parameters = 1.242134579 (5 changed)

**Comment by tobiscode** [comment]
Hi,
that's a very common question, so I address it here in the documentation in [FAQ 1: How do I know which penalties to use in `lasso_regression` and `ReweightingFunction`?](https://tobiscode.github.io/disstans/faq.html#how-do-i-know-which-penalties-to-use-in-subst)
TL;DR: These values are going to be dependent on your dataset and the type of analysis you want to make. Optimizing for an independent metric (e.g., the residuals of the secular velocity after estimating an Euler pole, see [Example 2](https://tobiscode.github.io/disstans/examples/example_2.html)) is what I usually do, using a grid search that comprises `eps`, `scale`, and initial `penalty`.
Cheers
